platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/LongListDisk.java

```diff
@@ -193,7 +193,7 @@ private ByteBuffer initOrGetTransferBuffer() {
     }
 
     static Path createTempFile(final String sourceFileName) throws IOException {
-        return LegacyTemporaryFileBuilder.buildTemporaryDirectory(STORE_POSTFIX).resolve(sourceFileName);
+        return Files.createTempDirectory(null).resolve(sourceFileName);
```

Comment:
I _think_ this should actually be using `FileSystemManager.resolveNewTemp(...)` so that temporary files are stored in the correct (not `/tmp`) location for merkldDB.

Reply:
Not sure how to properly initialize the file system manager e.g. using some config or inject is a dependency. 
Will need to spend more time on that. 🤔

Reply:
@jsync-swirlds [the last commit](https://github.com/hashgraph/hedera-services/pull/14695/commits/d4fa04a957be6480ba00027e3057b9f9f944d742) I added today is an attempt to resolve this comment (i.e. to use `FileSystemManager` in `LongListDisk`) and it feels hacky to me, perhaps @lpetrovic05 or @imalygin can advise if the approach makes sense.

Otherwise we can either revert to using `Files.createTempDirectory(null)` (if that's acceptable) or some alternative approach can be tried. Pleas let me know what you think.

Reply:
Looks reasonable to me.
One of our MerkleDB experts (e.g. @imalygin or @artemananiev) should probably check that the behavior of the MerkleDB remains correct.